src/test/java/org/testng/xml/XmlSuiteTest.java

```diff
@@ -57,4 +58,16 @@ public void testTimeOut(String timeout, int size, int lineNumber) throws IOExcep
         };
     }
 
+    @Test(description = "GITHUB-1668")
+    public void ensureNoExceptionsAreRaisedWhenMethodSelectorsDefinedAtSuiteLevel() throws IOException {
+        Parser parser = new Parser("src/test/resources/xml/issue1668.xml");
+        List<XmlSuite> suites = parser.parseToList();
+        XmlSuite xmlsuite = suites.get(0);
+        TestNG testNG = create();
+        testNG.setXmlSuites(suites);
+        testNG.setUseDefaultListeners(false);
+        testNG.run();
+        assertThat(xmlsuite.toXml()).isNotEmpty();
```

Comment:
The assertion is not really clear without a comment.
In fact, according to the issue, you just expect to not have an exception.

Reply:
Instead of just having a statement which invokes `xmlsuite.toXml()` (only when invoking this method the exception is raised, I chose to have an assertion that checks if the string is not empty. 

Yes you are correct, that a comment could have added value. I will amend the commit, add a comment and push the changeset